compiler/rustc_hir_typeck/src/expr_use_visitor.rs

```diff
@@ -367,6 +367,10 @@ impl<'tcx, Cx: TypeInformationCtxt<'tcx>, D: Delegate<'tcx>> ExprUseVisitor<'tcx
                 self.consume_exprs(args)?;
             }
 
+            hir::ExprKind::Use(expr, _) => {
+                self.consume_expr(expr)?;
```

Comment:
Hmm-- I just realized, this is.... not necessarily correct. Not necessarily *wrong* either. This says that `expr` will be *moved*, but the true behavior is that `expr` will be moved *unless* it implements `UseCloned`. So this should be adjusted later on to take that into account.

Reply:
Should we do something like `consume_copy_or_borrow`? if `UseCloned` borrow, if copy Copy and move otherwise.